x-pack/plugins/uptime/public/components/common/header/action_menu_content.tsx

```diff
@@ -57,7 +58,17 @@ export function ActionMenuContent({ config }: { config: UptimeConfig }): React.R
           time: { from: dateRangeStart, to: dateRangeEnd },
           breakdown: monitorId ? 'observer.geo.name' : 'monitor.type',
           reportDefinitions: {
-            'monitor.name': selectedMonitor?.monitor?.name ? [selectedMonitor?.monitor?.name] : [],
+            'monitor.name':
+              /*
+               * The second condition exists because we don't want to link users to an
+               * exploratory view that is filtered by a specific monitor when they're on
+               * the Uptime Overview page.
+               *
+               * This condition is intended to allow monitor filtering on all other pages.
+               */
```

Comment:
Comment is helpful but seems unnecessary :) 
I think code is self explanatory, No?

Reply:
I was on the fence about it. I told myself if the reviewer mentioned it I'd delete it 😉

Reply:
Done https://github.com/elastic/kibana/pull/122907/commits/baa460ecb7489294bed503de42027c0b59e5d4cc.